### PR TITLE
adding app.set and getting settings

### PR DESCRIPTION
**Pull request by scottnath**
Need to add/retrieve settings like in `[app.set](https://expressjs.com/en/api.html#app.set)`.

thank you!
Scott

### CODE REVIEW

**Comment by scottnath** [comment]
@masotime also added `req.body` and `res.status` with tests.

thanks!

File: lib/mockexpress.js [hunk 1]
```diff
@@ -51,6 +63,10 @@ var MockExpress = function(route) {
 				return getMap[property];
 			};
 
+      if (requestObject.get('body')) {
```
Comment (masotime) [inline lib/mockexpress.js]: This special case should not be necessary. The module doesn't enforce any rules on req (apart from `.query`, `.method`, `.url` as defined in the `invoke` function above. You can simply set `req.body` before calling `invoke`, ref - test below.

File: test/test.js [hunk 2]
```diff
@@ -184,6 +184,66 @@ describe('mockExpress', function() {
...
+		app.put('/test/body', function (req, res) {
+			assert.equal(typeof req.body, 'object');
+			res.status(201);
+		});
+
+		var assertionCallback = app.makeAssertionCallback(done, function(err, sideEffects) {
+			assert.equal(sideEffects.status, 201);
+		});
+
+		const req = app.makeRequest({ 'body': { foo: 'bar' } });
```
Comment (masotime) [inline test/test.js]: So you don't need to set `body` here.

File: test/test.js [hunk 3]
```diff
@@ -184,6 +184,66 @@ describe('mockExpress', function() {
...
+		app.makeResponse(assertionCallback);
+
+		app.invoke('put', '/test/body', req);
```
Comment (masotime) [inline test/test.js]: Before this statement, just have
```
req.body = { foo: 'bar' }
```

**Review by masotime — CHANGES_REQUESTED** [review]
Thanks for adding the features @scottnath, I really appreciate you taking the time to do it. The module's a little outdated (still in Node 0.x era when we're all working in ES2015+ these days).

I'm good with merging, but I'd request you remove the `body` feature - such specific functionality shouldn't be necessary and your test / use-case can still pass (ref: comments)

Thanks!

**Comment by masotime** [comment]
LGTM 👍

**Comment by scottnath** [comment]
Thanks @masotime!

**Pull request merged**